xen/lib/lifecycle.mli

```diff
@@ -0,0 +1,16 @@
+(*
+ * Copyright (c) 2015 Thomas Leonard <talex5@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+val await_shutdown : unit -> [`Poweroff | `Reboot] Lwt.t
```

Comment:
Well, it does says "when the domain is *asked* to shut down" - perhaps it could be renamed to `await_shutdown_request`.